tests/Feature/FavoritesControllerTest.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Feature;
+
+use App\Models\City;
+use App\Models\User;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Tests\TestCase;
+
+class FavoritesControllerTest extends TestCase
+{
+    use RefreshDatabase;
+
+    public function testCityCanBeAddedToFavorites(): void
+    {
+        $user = User::factory()->create();
+
+        $this->actingAs($user);
+
+        $city = City::factory()->create();
+
+        $response = $this->post("/favorites", ["city_id" => $city->id]);
+
+        $response->assertStatus(200);
+
+        $this->assertDatabaseHas("favorites", [
+            "user_id" => $user->id,
+            "city_id" => $city->id,
+        ]);
+    }
+
+    public function testCityCanBeRemovedFromFavorites(): void
```

Comment:
User can`t send non-existent city ID because button component renders only for existing in database cities and takes IDs from them.
Buttons render only for authorized users.

Reply:
Ok, if user cannot manipulate DOM elements via browser console, but via external client (e.g. Postman) you can send post request to this route.
I think this should be tested how app handle such request from non logged users.